core/src/main/java/com/sequenceiq/cloudbreak/rotation/executor/CMUserRotationExecutor.java

```diff
@@ -0,0 +1,95 @@
+package com.sequenceiq.cloudbreak.rotation.executor;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cluster.api.ClusterSecurityService;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.rotation.context.CMUserRotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationExecutor;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretLocationType;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretRotationException;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.secret.domain.RotationSecret;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+
+@Component
+public class CMUserRotationExecutor implements RotationExecutor<CMUserRotationContext> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CMUserRotationExecutor.class);
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackDtoService stackService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Override
+    public void rotate(CMUserRotationContext rotationContext) {
+        LOGGER.info("Starting rotation of CM user for resource {} by creating a new user.", rotationContext.getResourceCrn());
+        ClusterSecurityService clusterSecurityService = getClusterSecurityService(rotationContext);
+        RotationSecret userRotationSecret = secretService.getRotation(rotationContext.getUserSecret());
```

Comment:
getRotation can return RotationSecret even if the secret is not in a rotated state (so the original secret is not present in the backup). Although it should not happen I handled it in salt boot rotation and throw an error if that's the case, rather that moving forward and fail in whatever state. Would you consider it too?

Reply:
After it the user can retry the flow and vault will probably generate a new secret.